src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bonds exception for index "
+                    + index);
+        }
+        grow();
+        System.arraycopy(values, index, values, index + 1, size - index);
+        values[index] = value;
+        size++;
+    }
 
+    private void grow() {
+        T[] arrayWithNewCapacity = (T[]) new Object[getNewCapacity(values.length)];
+        System.arraycopy(values, 0, arrayWithNewCapacity, 0, values.length);
+        values = arrayWithNewCapacity;
+    }
```

Comment:
let's check first if our array is full and grow it only if there is no free space left. (There is no need to copy array every time). Than we should probably rename the method to `ensureCapacity` or smth like this. And lets place private methods after public

Reply:
@MaliukDaria Done, thanks!)